tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp

```diff
@@ -1146,6 +1188,61 @@ void AstNamedNode::DumpAttributes(AstDumper* dumper, DumpNodeOptions const& opti
         [](AstDumper* dumper, DumpNodeOptions const& options) { dumper->Newline(); });
   }
 }
+void AstNamedNode::DumpDocumentation(AstDumper* dumper, DumpNodeOptions const& options) const
+{
+  if (m_nodeDocumentation)
+  {
+    dumper->AddDocumentRangeStart();
+    if (options.NeedsLeftAlign)
+    {
+      dumper->LeftAlign();
+    }
+    dumper->InsertComment("/**");
+    {
+      DumpNodeOptions innerOptions{options};
+      innerOptions.NeedsLeftAlign = true;
+      innerOptions.NeedsLeadingNewline = true;
+      innerOptions.NeedsTrailingNewline = false;
+      m_nodeDocumentation->DumpNode(dumper, innerOptions);
+    }
+    dumper->Newline(); // We need to insert a newline here to ensure that the comment is properly
+                       // closed.
+    dumper->LeftAlign();
+    dumper->InsertComment(" */");
+    if (options.NeedsTrailingNewline)
+    {
+      dumper->Newline();
+    }
+    dumper->AddDocumentRangeEnd();
+  }
+}
+
+void AstNamedNode::DumpSourceComment(AstDumper* dumper, DumpNodeOptions const& options) const
+{
+  if (options.NeedsLeadingNewline)
+  {
+    dumper->Newline();
+  }
+  if (options.NeedsLeftAlign)
+  {
+    dumper->LeftAlign();
+  }
+  dumper->InsertComment("// ");
```

Comment:
InsertComment(string) w/o the "//".  Have the insertcomment method handle splitting the lines if they are too long and have the method handle inserting the "// " prefix.

If need be you could have an overload InsertComment(string stringToWrrite, bool noPrefix)

Reply:
The "Insert" (and "Add") APIs are "stupid" APIs which simply emit elements to the ApiView output. An "Insert" method inserts content which will normally be seen by the customer, a "Add" method inserts metadata.

This change includes some of the scaffolding needed to implement soft line-wrap functionality, but it's not complete (I need a way of marking a cursor and resetting to the cursor).

For simplicities sake, I chose to not build comment reflowing functionality into this code, instead I preserve the flow from clang (mostly - block comment directives are treated as being an inline type, which suppresses the leading crlf on a paragraph node).

This particular code handles inserting the source filename, which we don't ever want to flow.